File: cn/java-spring-demo/src/main/java/com/itangcent/springboot/demo/swagger/dto/SwaggerModel.java
package com.itangcent.springboot.demo.swagger.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

@ApiModel(description = "一个swagger模型示例")
public class SwaggerModel {
    
    @ApiModelProperty(value = "模型id", required = true, example = "1")
    private Long id = 1L;
    
    @ApiModelProperty(value = "模型名称", required = true, example = "demo")
    private String name = "demo";
    
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
} 

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/swagger/dto/SwaggerModel.kt
package com.itangcent.springboot.kotlin.demo.swagger.dto

import io.swagger.annotations.ApiModelProperty

/**
 * Swagger模型类
 * 用于演示Swagger注解的使用方法
 */
class SwaggerModel {

    /**
     * 字段A
     * 已废弃的测试字段
     */
    @ApiModelProperty(value = "字段A", required = true)
    @Deprecated("不再使用")
    var a: String? = null

}